src/main/java/qupath/ext/instanseg/core/InstanSegModel.java

```diff
@@ -251,8 +258,64 @@ public Optional<Integer> getNumChannels() {
         return getModel().flatMap(model -> Optional.of(extractChannelNum(model)));
     }
 
-    private static int extractChannelNum(BioimageIoSpec.BioimageIoModel model) {
-        int ind = model.getInputs().getFirst().getAxes().toLowerCase().indexOf("c");
+    /**
+     * Try to check the output tensors from the model spec.
+     * @return The output tensors if the model is downloaded, otherwise empty.
+     */
+    public Optional<List<OutputTensor>> getOutputs() {
+        return getModel().flatMap(model -> Optional.ofNullable(model.getOutputs()));
+    }
+
+    /**
+     * Try to check the output classes from the model spec.
+     * @return The output classes if the model is downloaded, and it's present, otherwise empty.
+     */
+    public List<String> getClasses() {
```

Comment:
I think `getClasses()` is too similar to `getClass()`, which has a completely different meaning.

How about `getOutputClasses()`?

Reply:
Sorry, this is actually unused and should've been removed.